great_expectations/expectations/expectation.py

```diff
@@ -331,23 +343,35 @@ def _atomic_prescriptive_template(
 
         styling: Optional[dict] = runtime_configuration.get("styling")
 
-        template_str = "$expectation_type(**$kwargs)"
+        expectation_type: str
+        expectation_kwargs: dict
+        if configuration:
+            expectation_type = configuration.expectation_type
+            expectation_kwargs = configuration.kwargs
+        elif result and result.expectation_config:
+            expectation_type = result.expectation_config.expectation_type
+            expectation_kwargs = result.expectation_config.kwargs
```

Comment:
Because we don't have an `else` clause, aren't these variables possibly unbounded?

Reply:
Yes, they are. If I change the `elif` to `else` I will get static type checking errors, because `result` is `Optional`. We have a patter whereby if a renderer fails, it will fall back to the "failure string" renderer. After thinking more about it, I could use an `else` and then `assert isinstance(result.expectation_config, ExpectationConfiguration)` that would solve the `mypy` errors and this renderer would still fail if both `ExpectationValidationResult` and `ExpectationConfiguration` are missing.

Reply:
If we are actually expecting a particular error path at runtime then I think a `mypy` ignore error is appropriate.

Reply:
Actually it sounds like this calls for an overload.
You're saying that even tho `result` & `configuration` are both optional. One of them must be provided correct?

Reply:
@NathanFarmer 
https://mypy.readthedocs.io/en/stable/more_types.html#function-overloading